subsys/bluetooth/host/conn_internal.h

```diff
@@ -14,17 +14,47 @@
 typedef enum __packed {
 	BT_CONN_DISCONNECTED,         /* Disconnected, conn is completely down */
 	BT_CONN_DISCONNECT_COMPLETE,  /* Received disconn comp event, transition to DISCONNECTED */
-	BT_CONN_CONNECTING_SCAN,      /* Central passive scanning */
-	BT_CONN_CONNECTING_AUTO,      /* Central connection establishment w/ filter */
-	BT_CONN_CONNECTING_ADV,       /* Peripheral connectable advertising */
-	BT_CONN_CONNECTING_DIR_ADV,   /* Peripheral directed advertising */
-	BT_CONN_CONNECTING,           /* Central connection establishment */
+
+	BT_CONN_CENTRAL_CONNECTING,   /* Central connection establishment */
+	/** Central scans for a device preceding establishing a connection to it.
+	 *
+	 * This can happen when:
+	 * - The application has explicitly configured the stack to connect to the device,
+	 *   but the controller resolving list is too small. The stack therefore first
+	 *   scans to be able to retrieve the currently used (private) address, resolving
+	 *   the address in the host if needed.
+	 * - The stack uses this connection context for automatic connection establishment
+	 *   without the use of filter accept list. Instead of immediately starting
+	 *   the initiator, it first starts scanning. This allows the application to start
+	 *   scanning while automatic connection establishment in ongoing.
+	 *   It also allows the stack to use host based privacy for cases where this is needed.
+	 */
+	BT_CONN_CENTRAL_SCAN_BEFORE_CONNECTING,
+
+	/** Central initiates a connection to a device in the filter accept list.
+	 *
+	 * For this type of connection establishment controller the initiator is started
```

Comment:
I don't understand this sentence.
Maybe "In this type of connection establishment, the controller's initiator role is started immediately.